skema/program_analysis/TS2CAST/ts2cast.py

```diff
@@ -0,0 +1,812 @@
+import json
+from typing import Any
+
+from tree_sitter import Language, Parser
+
+from skema.program_analysis.CAST2FN.cast import CAST
+from skema.program_analysis.CAST2FN.model.cast import (
+    Module,
+    SourceRef,
+    Assignment,
+    LiteralValue,
+    Var,
+    VarType,
+    Name,
+    Operator,
+    AstNode,
+    SourceCodeDataType,
+    ModelImport,
+    FunctionDef,
+    Loop,
+    Call,
+    ModelReturn,
+    ModelIf,
+)
+
+
+from skema.program_analysis.TS2CAST.variable_context import VariableContext
+from skema.program_analysis.TS2CAST.node_helper import NodeHelper
+from skema.program_analysis.TS2CAST.util import generate_dummy_source_refs, preprocess
+
+
+class TS2CAST(object):
+    def __init__(self, source_file_path: str):
+        # Initialize tree-sitter
+        self.tree_sitter_fortran = Language("build/my-languages.so", "fortran")
+
+        # We load the source code from a file
+        self.source = None
+        with open(source_file_path, "r") as f:
+            self.source = f.read()
+
+        # Set up tree sitter parser
+        self.parser = Parser()
+        self.parser.set_language(self.tree_sitter_fortran)
+        self.tree = self.parser.parse(bytes(self.source, "utf8"))
+
+        # CAST objects
+        self.module_name = None
+        self.source_file_name = source_file_path
+        self.module = Module()
+
+        # Walking data
+        self.variable_context = VariableContext()
+
+        self.node_helper = NodeHelper(source_file_path, self.source)
+        self.parse_dict = self.node_helper.parse_tree_to_dict(self.tree.root_node)
+        # print(json.dumps(self.parse_dict))
+
+        # Start visiting
+        self.run(self.parse_dict)
+
+        generate_dummy_source_refs(self.module)
+
+        # Create outer cast wrapping
+        self.out_cast = CAST([self.module], "Fortran")
+        # print(
+        #            json.dumps(
+        #               out_cast.to_json_object(), sort_keys=True, indent=None
+        #            )
+        #        )
+
+    def run(self, root: dict):
+        self.module.source_refs = root["source_refs"]
+        self.module.body = []
+        for child in root["children"]:
+            child_cast = self.visit(child)
+            if isinstance(child_cast, list):
+                self.module.body.extend(child_cast)
+            else:
+                self.module.body.append(child_cast)
+
+    def visit(self, node: dict):
+        match node["type"]:
```

Comment:
@vincentraymond-ua , `match` / `case` is a Python 3.10 language feature.  Unfortunately, the project doesn't yet support versions beyond 3.8.  Can you please convert this to `if` / `else` instead?   On a related note, I recommend converting your local develop environment to use Python 3.8.